scripts/in_container/run_signature_consistency_verify.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import inspect
+import sys
+
+import libcst as cst
+from in_container_utils import AIRFLOW_ROOT_PATH, console
+
+DECORATOR_OPERATOR_MAP = {
+    "kubernetes": "airflow.providers.cncf.kubernetes.operators.pod.KubernetesPodOperator",
+    "sensor": "airflow.sdk.bases.sensor.BaseSensorOperator",
+    "virtualenv": "airflow.providers.standard.operators.python.PythonVirtualenvOperator",
+    "branch_virtualenv": "airflow.providers.standard.operators.python.BranchPythonVirtualenvOperator",
+    # Add more here...
+}
```

Comment:
I wonder how we can prevent people from forgetting. Maybe a separate pre-commit hook that checks if a commit adds an entry to the provider config (`task-decorators` in `provider.yaml`) but does not also change this? Or maybe we should just pull the list of classes from `provider.yaml` directly instead? This can probably be explored in a second PR after this is merged.